pkg/k8s/pod_utils.go

```diff
@@ -20,6 +24,33 @@ func GetPodCondition(pod *corev1.Pod, conditionType corev1.PodConditionType) *co
 	return nil
 }
 
+// NamespaceIndexer returns indexer to index in data store using namespace
+func NamespaceIndexer() cache.Indexers {
+	indexer := map[string]cache.IndexFunc{}
+	indexer[Namespace] = func(obj interface{}) (strings []string, err error) {
+		return []string{obj.(*corev1.Pod).Namespace}, nil
+	}
+	return indexer
+}
+
+// NodeNameIndexer returns indexer to index in the data store using node name
+func NodeNameIndexer() cache.Indexers {
+	indexer := map[string]cache.IndexFunc{}
+	indexer[NodeNameSpec] = func(obj interface{}) (strings []string, err error) {
+		return []string{obj.(*corev1.Pod).Spec.NodeName}, nil
+	}
+	return indexer
+}
+
+// NSKeyIndexer is the key function to index the pod object using namespace/name
+func NSKeyIndexer(obj interface{}) (string, error) {
```

Comment:
this function `k8s.NSKeyIndexer` doesn't hint that it's specific to a pod yet the implementation is.
Let's do either below
1. change the function name to have pod
2. move this into another file like object_utils.go, and use `meta.Accessor(obj)` to decouple the implementation from pod, which can work for any k8s object. (preferred approach)

same comment for NamespaceIndexer above.